client/src/pages/Clusters/Clusters.tsx

```diff
@@ -99,49 +68,106 @@ const transformUpdatedAtDate = (cluster: ClusterSummary) => {
-          name: cluster.id,
-          accuracy: cluster.accuracy,
-          state: transformIsRunningLabel(cluster),
-          notificationChannels: transformNotificationChannelsToColumns(cluster),
-          updatedAt: transformUpdatedAtDate(cluster),
-        }),
-      );
-
-      setClusters(clusterRows);
-      setIsLoading(false);
-    } catch (e: unknown) {
-      console.error('Failed to fetch clusters', e);
-    }
-  };
-
-  useEffect(() => {
-    fetchClusters();
-  }, []);
-
-  const header = <HeaderWithIcon title={'Clusters'} />;
-
-  return (
-    <PageTemplate header={header}>
-      <SectionComponent title={'Clusters'} icon={<SVGIcon iconName='clusters-icon'/>}>
-        {isLoading && <div>Loading...</div>}
-        {!isLoading && clusters.length > 0 && (
-          <Table columns={columns} rows={clusters} />
-        )}
-        {!isLoading && clusters.length == 0 && (
-          <div>No registered clusters yet</div>
-        )}
-      </SectionComponent>
-    </PageTemplate>
-  );
+    const [clusters, setClusters] = useState<ClusterDataRow[]>([]);
+    const [isLoading, setIsLoading] = useState(true);
+    const navigate = useNavigate();
+
+    const handleNewReportOnDemand = (id: string) => {
+        navigate(`/reports/${id}/on-demand`);
+    };
+
+    const handleScheduledReport = (id: string) => {
+        navigate(`/reports/${id}/scheduled`);
+    };
+
+    const columns: Array<TableColumn<ClusterDataRow>> = [
+        {
+            header: 'Name',
+            columnKey: 'name',
+            customComponent: (row: ClusterDataRow) => (
+                <LinkComponent href="#" isRunning={row.state === 'ONLINE'}>
+                    {row.name}
+                </LinkComponent>
+            ),
+        },
+        {
+            header: 'Accuracy',
+            columnKey: 'accuracy',
+            customComponent: ({ accuracy }) => <UrgencyBadge label={accuracy} />,
+        },
+        {
+            header: 'Notification',
+            columnKey: 'notificationChannels',
+            customComponent: ({ notificationChannels }) => (
+                <Channels channels={notificationChannels} />
+            ),
+        },
+        {
+            header: 'Updated at',
+            columnKey: 'updatedAt',
+        },
+        {
+            header: 'Reports',
+            columnKey: 'actions',
+            customComponent: (row: ClusterDataRow) => (
+                <div className='clusters--button'>
+                    <ActionButton
+                        onClick={() => handleScheduledReport(row.name)}
+                        description="scheduled"
+                        color={ActionButtonColor.GREEN}
+                    />
+                    <ActionButton
+                        onClick={() => handleNewReportOnDemand(row.name)}
+                        description="on demand"
+                        color={ActionButtonColor.GREEN}
+                    />
+                </div>
```

Comment:
Consider creating custom component

Reply:
Do you think it's nececery for that purpose?